src/main/java/io/cdap/plugin/gcp/dataplex/sink/DataplexOutputFormatProvider.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.dataplex.sink;
+
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.validation.FormatContext;
+import io.cdap.cdap.etl.api.validation.ValidatingOutputFormat;
+import io.cdap.plugin.gcp.bigquery.sink.BigQueryOutputFormat;
+import io.cdap.plugin.gcp.bigquery.util.BigQueryConstants;
+import io.cdap.plugin.gcp.bigquery.util.BigQueryUtil;
+import io.cdap.plugin.gcp.dataplex.common.util.DataplexConstants;
+import io.cdap.plugin.gcp.gcs.sink.GCSOutputFormatProvider;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.OutputCommitter;
+import org.apache.hadoop.mapreduce.OutputFormat;
+import org.apache.hadoop.mapreduce.RecordWriter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+/**
+ * OutputFormatProvider for Dataplex
+ */
+public class DataplexOutputFormatProvider implements ValidatingOutputFormat {
+  public static final String RECORD_COUNT_FORMAT = "recordcount.%s";
+  public static final String DATAPLEX_ASSET_TYPE = "dataplexsink.assettype";
+  public static final String DATAPLEX_OUTPUT_BASE_DIR = "dataplex.output.fileoutputformat.baseoutputdir";
+  private static final String DELEGATE_OUTPUTFORMAT_CLASSNAME = "gcssink.delegate.outputformat.classname";
```

Comment:
This is also defined here - https://github.com/data-integrations/google-cloud/blob/develop/src%2Fmain%2Fjava%2Fio%2Fcdap%2Fplugin%2Fgcp%2Fgcs%2Fsink%2FGCSOutputFormatProvider.java#L23
Can we dedupe?

Reply:
yes